src/Compilers/Core/Portable/MetadataReader/MetadataDecoder.cs

```diff
@@ -319,6 +319,24 @@ private TypeSymbol DecodeTypeOrThrow(ref BlobReader ppSig, SignatureTypeCode typ
                     typeSymbol = DecodeGenericTypeInstanceOrThrow(ref ppSig, out refersToNoPiaLocalType);
                     break;
 
+                case SignatureTypeCode.FunctionPointer:
+                    var signatureHeader = ppSig.ReadSignatureHeader();
+                    var parameters = DecodeSignatureParametersOrThrow(ref ppSig, signatureHeader, typeParameterCount: out int typeParamCount, shouldProcessAllBytes: false);
+                    Debug.Assert(typeParamCount == 0);
+                    Cci.CallingConvention callingConvention = signatureHeader.CallingConvention switch
+                    {
+                        SignatureCallingConvention.Default => Cci.CallingConvention.Default,
+                        SignatureCallingConvention.CDecl => Cci.CallingConvention.CDecl,
+                        SignatureCallingConvention.FastCall => Cci.CallingConvention.FastCall,
```

Comment:
>Cci.CallingConvention.FastCall [](start = 63, length = 30)

It looks like we will not be able to serialize a type like that. Is there a reason for that? In general, we should be able to gracefully handle everything we import, but it looks like MetadatWriter is going to crash on this convention.

Reply:
As currently spec'd, C# function pointers don't have fastcall. I'll raise the question and make this throw for now.

---
In reply to: [374388482](https://github.com/dotnet/roslyn/pull/41340#discussion_r374388482) [](ancestors = 374388482)